ui/components-react/src/components-react/propertygrid/component/VirtualizedPropertyGridWithDataProvider.tsx

```diff
@@ -47,25 +47,35 @@ export interface VirtualizedPropertyGridWithDataProviderProps
 export function VirtualizedPropertyGridWithDataProvider(
   props: VirtualizedPropertyGridWithDataProviderProps
 ) {
-  const modelSource = usePropertyGridModelSource({
+  const [showSpinner, setShowSpinner] = useState(false);
+
+  const { modelSource, inProgress } = useTrackedPropertyGridModelSource({
     dataProvider: props.dataProvider,
   });
+
   const model = usePropertyGridModel({ modelSource });
   const eventHandler = usePropertyGridEventHandler({ modelSource });
+  useEffect(() => {
+    const timeout = setTimeout(() => {
```

Comment:
I think if `inProgress` is `false` we don't need a delay before updating `showSpinner` flag.